### PR TITLE
[GLUTEN-1902] Separate the implementations for TranslateTransformer

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

As CH has some limitations for supporting translate function, but velox not, this PR proposes to separate the implementations.

## How was this patch tested?

Existing spark UTs.

### CODE REVIEW

**Comment by PHILO-HE** [comment]
Blocked by upstream & rebase.

File: backends-clickhouse/src/main/scala/io/glutenproject/expression/CHExpressionTransformer.scala [hunk 1]
```diff
@@ -335,6 +335,39 @@ case class CHTruncTimestampTransformer(
...
+case class CHStringTranslateTransformer(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    matchingExpr: ExpressionTransformer,
+    replaceExpr: ExpressionTransformer,
+    original: StringTranslate)
+  extends ExpressionTransformer {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    // In CH, translateUTF8 requires matchingExpr and replaceExpr argument have the same length
```
Comment (PHILO-HE) [inline backends-clickhouse/src/main/scala/io/glutenproject/expression/CHExpressionTransformer.scala:347]: @exmy, could you take a review? I am assuming CH still have this limitation. cc @zzcclp

Comment (exmy) [reply backends-clickhouse/src/main/scala/io/glutenproject/expression/CHExpressionTransformer.scala:347]: yes, CH translateUTF8 has this limitation.

**Review by JkSelf — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**